### ISSUE TITLE
FlexibleConnectonPositionStrategy _pushOverlayOnScreen doesn't take position: fixed into account when the page is scrolled

### ISSUE THREAD

**Issue by jpzwarte** [angular/material2#11763]
#### Bug, feature request, or proposal:
So i have a custom component (a popover) that uses CdkOverlay to display a popover anchored to an origin element. The origin element is part of a position: fixed page-header. What happens is that the overlay initially doesn't fit in the narrowed viewport (doesn't fit on the x-axis). It then tries to push the overlay on screen using ```_pushOverlayOnScreen```. While that fixes the x-axis issue, it also *breaks* the y-axis position.

Since the page is scrolled, it calculates the ```overflowTop``` value using the scroll position and then *incorrectly* adds that value to the y position. This can cause the popover to appear halfway down the page when the page is scrolled significantly.

#### What is the expected behavior?
It should detect that the origin (parent) element is ```position: fixed``` and not take the scroll position into account?

#### Which versions of Angular, Material, OS, TypeScript, browsers are affected?
Angular 6.0.3
Cdk 6.2.0

**Comment by jpzwarte** [comment]
I've added a hack/workaround in my popover component to adjust "top":

```
positionStrategy.positionChanges.subscribe(change => {
  const origin = this._containerInstance._config.origin,
        originElement = origin instanceof ElementRef ? origin.nativeElement : origin;

  if (getComputedStyle(originElement.offsetParent).position === 'fixed') {
    this._overlayRef.overlayElement.style.top = originElement.offsetParent.clientHeight + 'px';
  }
})
```

**Comment by FDIM** [comment]
I can reproduce something similar with v6.3.3 - seems like there is an issue in `_pushOverlayOnScreen` method that takes top scroll offset into account when it shouldn't.

Also origin element (or any parent) is not using position:fixed in our case. 

![image](https://user-images.githubusercontent.com/1985429/42756909-59a0b7b4-88fe-11e8-8d8a-2713aa07930a.png)

If I add breakpoint before `return` statement in the image and change pushY to 0 - position ends up correct

**Comment by jackhkmatthews** [comment]
Experiencing the same issue as FDIM.

**Comment by crisbeto** [comment]
This has been fixed in the 7.0 beta by https://github.com/angular/material2/pull/12624.
